android/src/main/java/sqip/flutter/SquareInAppPaymentsFlutterPlugin.java

```diff
@@ -28,140 +30,121 @@
-    // channel.setMethodCallHandler(this);
-    // cardEntryModule = new CardEntryModule(channel);
-    // googlePayModule = new GooglePayModule(channel);
+  public void onAttachedToEngine(FlutterPluginBinding binding) {
+    this.applicationContext = binding.getApplicationContext();
+    channel = new MethodChannel(binding.getBinaryMessenger(), "square_in_app_payments");
+    // MethodCallHandler is set in onAttachedToActivity to avoid NPE issues
   }
 
   @Override
-  public void onDetachedFromEngine(FlutterPluginBinding flutterPluginBinding) {
+  public void onDetachedFromEngine(FlutterPluginBinding binding) {
     cardEntryModule = null;
     googlePayModule = null;
     channel = null;
   }
 
   @Override
-  public void onAttachedToActivity(ActivityPluginBinding activityPluginBinding) {
-    // KNOWN ISSUE: OnAttachedToEngine can be called twice which may be due to https://github.com/flutter/flutter/issues/69721
-    // Once the issue is resolved, we can check if we can move following three lines to OnAttachedToEngine.
+  public void onAttachedToActivity(ActivityPluginBinding activityBinding) {
     channel.setMethodCallHandler(this);
-    cardEntryModule = new CardEntryModule(channel);
-    googlePayModule = new GooglePayModule(channel);
-    googlePayModule.attachActivityResultListener(activityPluginBinding, channel);
-    cardEntryModule.attachActivityResultListener(activityPluginBinding, channel);
+    cardEntryModule = new CardEntryModule(applicationContext, channel);
+    googlePayModule = new GooglePayModule(applicationContext, channel);
+
+    googlePayModule.attachActivityResultListener(activityBinding, channel);
+    cardEntryModule.attachActivityResultListener(activityBinding, channel);
   }
 
   @Override
-  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding activityPluginBinding) {
-    // KNOWN ISSUE: OnAttachedToEngine can be called twice which may be due to https://github.com/flutter/flutter/issues/69721
-    // Once the issue is resolved, we can check if we can move following three lines to OnAttachedToEngine.
-    channel.setMethodCallHandler(this);
-    cardEntryModule = new CardEntryModule(channel);
-    googlePayModule = new GooglePayModule(channel);
-    googlePayModule.attachActivityResultListener(activityPluginBinding, channel);
-    cardEntryModule.attachActivityResultListener(activityPluginBinding, channel);
+  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding activityBinding) {
+    onAttachedToActivity(activityBinding);
   }
 
-
   @Override
   public void onDetachedFromActivityForConfigChanges() {
-    // KNOWN ISSUE: OnAttachedToEngine can be called twice which may be due to https://github.com/flutter/flutter/issues/69721
-    // Once the issue is resolved, we can check if we can remove following three lines.
     cardEntryModule = null;
     googlePayModule = null;
     channel = null;
```

Comment:
Do we still need these three lines? Is the issue resolved that required this workaround?